addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1866,9 +1866,24 @@ var SnippetsMenu = Widget.extend({
         this._addToolbar();
         this._checkEditorToolbarVisibilityCallback = this._checkEditorToolbarVisibility.bind(this);
         $(this.options.wysiwyg.odooEditor.document.body).on('click', this._checkEditorToolbarVisibilityCallback);
+        this.invisibleDOMPanelEl = document.createElement('div');
+        this.invisibleDOMPanelEl.classList.add('o_we_invisible_el_panel');
+        this.invisibleDOMPanelEl.appendChild(
+            $('<div/>', {
+                text: _t('Invisible Elements'),
+                class: 'o_panel_header',
+            })[0]
+        );
+
+        // Prepare snippets editor environment
+        this.$snippetEditorArea = $('<div/>', {
+            id: 'oe_manipulators',
+        });
+        this.$body.prepend(this.$snippetEditorArea);
 
         if (this.options.enableTranslation) {
             // Load the sidebar with the style tab only.
+            defs.push(this._updateInvisibleDOM());
             await this._loadSnippetsTemplates();
```

Comment:
@qsm-odoo , after reflexion, I am wondering if those two lines should not be inverted (like this : )
```js
await this._loadSnippetsTemplates();
defs.push(this._updateInvisibleDOM());
```
In order to, as in other location on the code, first execute `_loadSnippetsTemplates()` before `_updateInvisibleDOM()` . Indeed, `_createSnippetEditor()` is called in  `_updateInvisibleDOM()` and this function uses variables (for example `this.templateOptions`) that are set thanks to  `_loadSnippetsTemplates()`.

Reply:
I am rather wondering why `_loadSnippetsTemplates` is called in translate mode actually? :thinking: Do you know why?

Reply:
It think that before this commit, `_loadSnippetsTemplates()` was called to build the right sidebar (the snippets menu) (see https://github.com/odoo/odoo/blame/081021b1166d8c61b07132cb512870c30f11f7cf/addons/web_editor/static/src/js/editor/snippets.editor.js#L2451 ) and have the "Style" tab in order to be able to change the font of the text.